### PR TITLE
Implements get_enabled_features api

### PR DESCRIPTION
**Pull request by rashidsp**

### CODE REVIEW

File: lib/optimizely.rb [hunk 1]
```diff
@@ -282,6 +282,28 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
       false
     end
 
+    def get_enabled_features(user_id, attributes = nil)
+      # Returns the list of features that are enabled for the user.
```
Comment (oakbani) [inline lib/optimizely.rb]: Get keys of all feature flags which are enabled for the user

File: lib/optimizely.rb [hunk 2]
```diff
@@ -282,6 +282,28 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
...
+      # Args:
+      #   user_id: ID for user.
+      #   attributes: Dict representing user attributes.
+      # Returns:
+      #   A list of the keys of the features that are enabled for the user.
```
Comment (oakbani) [inline lib/optimizely.rb]: Just a suggestion: List of feature flag keys

File: lib/optimizely.rb [hunk 3]
```diff
@@ -282,6 +282,28 @@ def is_feature_enabled(feature_flag_key, user_id, attributes = nil)
...
+      enabled_features = []
+
+      unless @is_valid
+        logger = SimpleLogger.new
+        logger.log(Logger::ERROR, InvalidDatafileError.new('get_enabled_features').message)
+        return enabled_features
+      end
+
+      @config.feature_flags.each do |feature|
+        enabled_features.push(feature['key']) if is_feature_enabled(feature['key'], user_id, attributes)
```
Comment (oakbani) [inline lib/optimizely.rb]: Please only push feature key when is_feature_enabled returns strictly 'true'. 
FYI - In Ruby only false and nil are falsey. Everything else is truthy (yes, even 0 is truthy).

File: spec/project_spec.rb [hunk 4]
```diff
@@ -703,6 +703,37 @@ class InvalidErrorHandler; end
...
+    end
+
+    it 'should return only enabled feature flags keys' do
+      # Sets all feature-flags keys with randomly assigned status
+      features_keys = project_instance.config.feature_flags.map do |item|
+        {key: (item['key']).to_s, value: [true, false].sample} # '[true, false].sample' generates random boolean
+      end
+
+      enabled_features = features_keys.map { |x| x[:key] if x[:value] == true }.compact
+      prevented_features = features_keys.map { |x| x[:key] if x[:value] == false }.compact
```
Comment (oakbani) [inline spec/project_spec.rb]: change prevented to disabled

File: lib/optimizely.rb (hunk 1 above)
Comment (rashidsp) [reply lib/optimizely.rb]: Done

File: lib/optimizely.rb (hunk 2 above)
Comment (rashidsp) [reply lib/optimizely.rb]: Done

File: spec/project_spec.rb (hunk 4 above)
Comment (rashidsp) [reply spec/project_spec.rb]: Done

**Pull request closed without merging**